### DISCUSSION TITLE
How to download an old version of cURL?

### DISCUSSION THREAD

**Discussion by IonBoleac** [curl/curl-for-win#71] [Q&A] [answered]
Hello everyone,
I want to use curl on Windows XP Pro SP3 and for that, I must use the Win32 application version of cURL. Do you know where I can download this specific version? 

Thanks for helping me

**Comment by vszakats** [comment, marked as answer]
7.84.0 from around 2022-08-22 was the latest of these builds with "official" XP support. Have you looked for `https://curl.se/windows/` on `archive.org`? E.g.: https://web.archive.org/web/20220828191750/https://curl.se/windows/ ?

That is a fairly old version with known issues. We recommend building the latest from source, if possible.

edit: this build needs the Universal UCRT package to be installed manually. For Windows XP this is v14.27.29114.0, avaiable from Microsoft:
- x86: https://download.visualstudio.microsoft.com/download/pr/56f631e5-4252-4f28-8ecc-257c7bf412b8/D305BAA965C9CD1B44EBCD53635EE9ECC6D85B54210E2764C8836F4E9DEFA345/VC_redist.x86.exe
- x64: https://download.visualstudio.microsoft.com/download/pr/722d59e4-0671-477e-b9b1-b8da7d4bd60b/591CBE3A269AFBCC025681B968A29CD191DF3C6204712CBDC9BA1CB632BA6068/VC_redist.x64.exe

**Reply by IonBoleac** [reply]
Thanks for your cordial answer. 

I need only a working version on Windows XP to test it. For that, I tried this version https://archive.org/details/generic-curl-7-65-1_3-win32-mingw but when I run curl the application tells me that is missing the file Normaliz.dll. Do you know a version that might work?


But how can I build a new version for Win 32 from the source code?

**Reply by vszakats** [reply]
This is a very old build. Have you tried the one linked from the page in my message? That one should not require `Normaliz.dll`.

For building from source, this document is a good starting point: https://github.com/curl/curl/blob/master/docs/INSTALL-CMAKE.md

**Reply by IonBoleac** [reply]
I tried the version that you linked but windows xp tells me that is not a win32 application. I tried a lot of versions and all of them tells me that are non available for win32

**Reply by vszakats** [reply]
Is this happening with this specific download, too?:
https://web.archive.org/web/20220828191750/https://curl.se/windows/dl-7.84.0_9/curl-7.84.0_9-win32-mingw.zip

**Reply by IonBoleac** [reply]
Yeah, i tried exactly this version and when i run that curl version the machine response with "not valide win32 application"

**Reply by vszakats** [reply]
Thanks! ~~This suggests XP support was broken even before officially dropping support for it.~~ edit: this might be missing UCRT libs: https://github.com/curl/curl-for-win/discussions/71#discussioncomment-9451365

If you can get hold of the official IDN package for XP ("Microsoft Internationalized Domain Name (IDN) Mitigation APIs"), it can solve the missing `Normaliz.dll` issue, but Microsoft deleted that download from their website.

**Reply by IonBoleac** [reply]
Nice!!! BTW thanks to helping me

**Comment by killerbees19** [comment]
Afaik cURL v7.80.0 (from 2021-11-10!) was the last official release that works at Windows XP. I've tested many versions in 2023...

You can download it here: https://web.archive.org/web/20211208160135/https://curl.se/windows/

**Reply by GenesisFR** [reply]
Can confirm 7.80 worked out-of-the-box on Windows XP SP3, make sure to include `curl-ca-bundle.crt` or it'll fail.

**Comment by vszakats** [comment]
Found a clue that may make 7.84.0 work on XP. That release used UCRT and UCRT is a separate install for Windows XP, as depicted by the curl-for-win README from that time:
https://github.com/curl/curl-for-win/blob/f4c37bb0b34fd385bff9c6a65ed1a4776c046f65/README.md

XP needs v14.27.29114.0 of the Universal UCRT package, installed manually:
- x86: https://download.visualstudio.microsoft.com/download/pr/56f631e5-4252-4f28-8ecc-257c7bf412b8/D305BAA965C9CD1B44EBCD53635EE9ECC6D85B54210E2764C8836F4E9DEFA345/VC_redist.x86.exe
- x64: https://download.visualstudio.microsoft.com/download/pr/722d59e4-0671-477e-b9b1-b8da7d4bd60b/591CBE3A269AFBCC025681B968A29CD191DF3C6204712CBDC9BA1CB632BA6068/VC_redist.x64.exe

**Comment by flawrist** [comment]
Just grabbed older versions of Curl for Windows from
* 6.0 … 7.33 https://ftp.sunet.se/mirror/archive/ftp.sunet.se/pub/www/utilities/curl/ 
* 7.34 http://www.math.kobe-u.ac.jp/Asir/Add-ons/
* 7.53.1 https://winampplugins.co.uk/curl/
* the rest https://community.chocolatey.org/packages/curl/#versionhistory

Checked for viruses with [Jotti](https://virusscan.jotti.org/) and [VirusTotal](https://www.virustotal.com/old-browsers/) online scanners before running.

**Reply by vszakats** [reply]
> FYI: Just grabbed Curl 7.11 and 7.12 for Windows https://ftp.sunet.se/mirror/archive/ftp.sunet.se/pub/www/utilities/curl/

For the record, binaries distributed there were not built by this project.

edit: nor the other ones added to the message later.

**Reply by bagder** [reply]
Downloading random binaries from the internet and trusting them to be what they claim to be...
